### DISCUSSION TITLE
Default shortened URLs (ozh, yourls, yourlsblog) work, but any new shortened URLs only direct back to the root subdomain

### DISCUSSION THREAD

**Discussion by EscoNomad** [YOURLS/YOURLS#3487] [Q&A] [answered]
I'm sorry if this has been covered- I've been through all the documentation and searched for a relevant discussions but didn't see something related to this issue. Unfortunately, I'm not sure what to use for the search, and most of the words are very common (redirect, url, domain, etc).

I've installed Yourls on a subdomain (x.organicscience.jp), and everything appears to be functioning correctly. I was able to get through the install process and the interface is working properly.

The three default shortened URLs- ozh, yourls, yourlsblog- all work as intended. However, when I try to add a new entry, everything looks correct in the UI, but the URL redirects the browser to x.organicscience.jp. 

Using the URL will only return me to that URL. I am unable to load the site I am shortening.

I have checked the .htaccess file, and it appears to be correct. Also, the config.php seems to be set up correctly.

I am using Dreamhost and have installed Yourls on a different hosted domain, and it worked without issue.

I thought it may be due to a lack of index file, but upon adding a blank index file, it still functioned in the same way.

An example:

https://x.organicscience.jp/fb - Should redirect to facebook.com
https://x.organicscience.jp/ozh - works correctly

**Comment by dgw** [comment, marked as answer]
I had a theory that caching was involved, but since you provided test links and the exact same behavior happened to me… Nope, that theory was immediately proved wrong. But testing led me to what _is_ happening.

The default links have a scheme: For example, `ozh` points to `https://ozh.org/`.

Your test `fb` keyword points to `facebook.com`. It has no scheme. When YOURLS sends this in a `Location` header, this value is treated as a relative URL—the browser tries to load `https://x.organicscience.jp/facebook.com`, and is then redirected to `/` (the root) by YOURLS' default keyword-not-found behavior.

Here is a verbose cURL log, with the irrelevant bits (TLS negotiation, raw IPs, etc.) stripped out, showing what is happening:

```
$ curl -Lqv https://x.organicscience.jp/fb
[... IP finding ...]
* Connected to x.organicscience.jp (***) port 443 (#0)
[... TLS negotiation ...]
> GET /fb HTTP/2
> Host: x.organicscience.jp
> user-agent: curl/7.85.0
> accept: */*
>
[... More TLS ...]
< HTTP/2 301
< date: Wed, 28 Dec 2022 11:03:31 GMT
< server: Apache
< x-robots-tag: noindex
< location: facebook.com
< cache-control: max-age=600
< expires: Wed, 28 Dec 2022 11:13:31 GMT
< vary: User-Agent
< content-length: 0
< content-type: text/html; charset=UTF-8
<
* Connection #0 to host x.organicscience.jp left intact
* Issue another request to this URL: 'https://x.organicscience.jp/facebook.com'
[... More connection & TLS management ...]
> GET /facebook.com HTTP/2
> Host: x.organicscience.jp
> user-agent: curl/7.85.0
> accept: */*
>
* TLSv1.2 (IN), TLS header, Supplemental data (23):
< HTTP/2 302
< date: Wed, 28 Dec 2022 11:03:31 GMT
< server: Apache
< location: https://x.organicscience.jp
< cache-control: max-age=600
< expires: Wed, 28 Dec 2022 11:13:31 GMT
< vary: User-Agent
< content-length: 0
< content-type: text/html; charset=UTF-8
<
* Connection #0 to host x.organicscience.jp left intact
* Issue another request to this URL: 'https://x.organicscience.jp/'
[... So much TLS junk! ...]
> GET / HTTP/2
> Host: x.organicscience.jp
> user-agent: curl/7.85.0
> accept: */*
>
* TLSv1.2 (IN), TLS header, Supplemental data (23):
< HTTP/2 200
< date: Wed, 28 Dec 2022 11:03:31 GMT
< server: Apache
< last-modified: Wed, 28 Dec 2022 09:54:58 GMT
< etag: "1a3-5f0e05c74b033"
< accept-ranges: bytes
< content-length: 419
< cache-control: max-age=600
< expires: Wed, 28 Dec 2022 11:13:31 GMT
< vary: Accept-Encoding,User-Agent
< content-type: text/html
<
* TLSv1.2 (IN), TLS header, Supplemental data (23):
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd">
<html>
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8">
  <meta http-equiv="Content-Style-Type" content="text/css">
  <title></title>
  <meta name="Generator" content="Cocoa HTML Writer">
  <meta name="CocoaVersion" content="2299">
  <style type="text/css">
  </style>
</head>
<body>
</body>
</html>
* Connection #0 to host x.organicscience.jp left intact
```

**Reply by EscoNomad** [reply]
I see!

The simplest solution, but it totally eluded me.

Thank you for your quick help and support- adding the scheme worked perfectly.

I hope you and the team at Yourls have a great New Year. Thank you for all the support!
